src/sampler_learning.py

```diff
@@ -85,10 +85,14 @@ def _create_sampler_data(
     """
     positive_data = []
     negative_data = []
-    for idx, part_transitions in enumerate(transitions):
-        for ((state, _, _, option, _, trans_add_effects, trans_delete_effects),
-             obj_to_var) in part_transitions:
-            assert option.parent == param_option
+    for idx, partition in enumerate(partitions):
+        for (segment, obj_to_var) in partition:
+            traj, option, before, after = segment
+            state = traj[0][0]
+            trans_add_effects = after - before
+            trans_delete_effects = before - after
+            if option.parent != param_option:
```

Comment:
Why was this assert from before changed to a if not: continue?

Reply:
because before, we had already grouped partitions into options; that's not the case anymore